Fix:

libtool: compile: unable to infer tagged configuration
libtool:   error: specify a tag with '--tag'

https://github.com/net-snmp/net-snmp/issues/647
https://github.com/net-snmp/net-snmp/pull/648
--- Makefile.top.orig	2023-08-15 15:32:01.000000000 -0500
+++ Makefile.top	2023-08-29 01:38:51.000000000 -0500
@@ -86,18 +86,18 @@
 LIBAGE      = 2
 LIBREVISION = 1
 
-LIB_LD_CMD      = $(LIBTOOL) --mode=link $(LINKCC) $(CFLAGS) -rpath $(libdir) -version-info $(LIBCURRENT):$(LIBREVISION):$(LIBAGE) @LD_NO_UNDEFINED@ -o
+LIB_LD_CMD      = $(LIBTOOL) --mode=link --tag=CC $(LINKCC) $(CFLAGS) -rpath $(libdir) -version-info $(LIBCURRENT):$(LIBREVISION):$(LIBAGE) @LD_NO_UNDEFINED@ -o
 LIB_EXTENSION   = la
 LIB_VERSION     =
 LIB_LDCONFIG_CMD = $(LIBTOOL) --mode=finish $(INSTALL_PREFIX)$(libdir)
-LINK		= $(LIBTOOL) --mode=link $(LINKCC)
+LINK		= $(LIBTOOL) --mode=link --tag=CC $(LINKCC)
 # RANLIB 	= @RANLIB@
 RANLIB		= :
 
 # libtool definitions
 .SUFFIXES: .c .o .lo .rc
 .c.lo:
-	$(LIBTOOL) --mode=compile $(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $<
+	$(LIBTOOL) --mode=compile --tag=CC $(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $<
 .rc.lo:
 	$(LIBTOOL) --mode=compile --tag=RC @RC@ -o $@ -i $<
 
